packages/client/hmi-client/src/services/data.ts

```diff
@@ -568,17 +575,18 @@ const fetchData = async (
 			// are we executing a search-by-example
 			// (i.e., to find similar documents or related artifacts for a given document)?
 			if (searchParam.xdd && searchParam?.xdd.dataset) {
-				if (searchParam?.xdd.similar_search_enabled) {
-					const relatedDocuments = await getRelatedDocuments(
-						searchParam?.xdd.related_search_id as string,
-						searchParam?.xdd.dataset
-					);
-					const similarDocumentsSearchResults = {
-						results: relatedDocuments,
-						searchSubsystem: ResourceType.XDD
-					};
-					finalResponse.allData.push(similarDocumentsSearchResults);
-					finalResponse.allDataFilteredWithFacets.push(similarDocumentsSearchResults);
+				if (searchParam?.xdd.similar_search_enabled && searchParam?.xdd.related_search_id) {
+					const response = await fetchResource('', resourceType as ResourceType, {
+						...searchParamWithFacetFilters,
+						xdd: {
+							...searchParamWithFacetFilters?.xdd,
+							similar_to: searchParam.xdd.related_search_id as string,
+							max: 100,
```

Comment:
Is this returning the search results to 100? I believe we had reduced the amount to 20 for performance reasons unless you're finding the unified endpoint now fast enough with Matt's changes

Reply:
Sounds good!